packages/popover/src/vaadin-popover.js

```diff
@@ -787,7 +714,10 @@ class Popover extends PopoverPositionMixin(
       return;
     }
 
-    if ((this.__hasTrigger('focus') && this.__mouseDownInside) || this._overlayElement.contains(event.relatedTarget)) {
```

Comment:
Changed `this._overlayElement.contains` with `this.contains` here and elsewhere which seems correct: in case of moving focus by the browser <kbd>Tab</kbd> logic, the `relatedTarget` is `vaadin-popover` itself.